### ISSUE TITLE
[Idea] provide some form of scaffolding

### ISSUE THREAD

**Issue by woutervb** [canonical/operator#224]
As with many (new) frameworks that expose a lot of functionality, it's is easy to shoot oneself in the foot.
This is something that cannot be prevented, but I think that many 'beginners' mistakes could be prevented by providing some templates on how to realize certain functionality.

E.g. some helping code to create an operator framework 'project' with all links in place, using some templates for e.g. installing a snap, or a deb, or creating an interface to be reused.

**Comment by chipaca** [comment]
Yes, we will be providing something like that, not in the operator framework itself but via the (new, TBD) charm tool.
